frontend/src/modules/Dashboard/Components/CourseTable.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import Table from '@mui/material/Table'
+import {
+  StyledTextBox,
+  StyledSmallText,
+  StyledClassesContainer,
+  StyledNoClasses,
+} from 'Dashboard/Styles/Groups.style'
+
+import {
+  Box,
+  Paper,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from '@mui/material'
+import { Course } from '@core/Types'
+import { useHistory } from 'react-router-dom'
+import Link from '@mui/material/Link'
+import { EDIT_ZING_PATH } from '@core/Constants'
+import { defaultSortingOrder, defaultFilterOption } from './Dashboard'
+
+export const CourseTable = ({ courses }: CourseGridProps) => {
+  const history = useHistory()
+
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        overflow: 'auto',
+      }}
+    >
+      {courses.length === 0 ? (
+        <>
+          <StyledTextBox>
+            <h2>No Courses to Show</h2>
+          </StyledTextBox>
+          <StyledSmallText>
+            Once students request study partners, they'll automatically be
+            placed into courses on this page!
+          </StyledSmallText>
+          <StyledClassesContainer>
+            {[...Array(8)].map((_, i) => (
+              <StyledNoClasses key={i} />
+            ))}
+          </StyledClassesContainer>
+        </>
+      ) : (
+        <Box
+          sx={{
+            width: '100%',
+            justifyContent: 'center',
+            gap: 4,
+            gridTemplateColumns: 'repeat(auto-fill, 300px)',
```

Comment:
I don't think that the lines `gap: 4` and `gridTemplateColumns: 'repeat(auto-fill, 300px)` are contributing to the styling here since you are styling a table and not a grid, so I think that these lines can be removed without changing the look of the content.